### PR TITLE
Branch 158779483

### PR DESCRIPTION
**Pull request by martinwicke**
Complex merge in FFT ops, Estimator tests, LMDB reader.

### CODE REVIEW

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Comment by martinwicke** [comment]
Ah, thanks, Jenkins. You're totally on top of it.

**Review by jhseu — APPROVED** [review]

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Comment by martinwicke** [comment]
@rryan @ispirmustafa LMK if my merge messed with your code.

**Comment by caisq** [comment]
The tensorboard-related failures in Linux CPU Test (Python 3) are unrelated and pending fix internally. Merging PR.

File: tensorflow/python/kernel_tests/fft_ops_test.py [hunk 1]
```diff
@@ -264,142 +296,155 @@ def _tfIFFTForRank(self, rank):
...
-            self._CompareForward(r2c.astype(np.float32), rank, fft_length)
-            self._CompareBackward(c2r.astype(np.complex64), rank, fft_length)
-
-            # Confirm it works with unknown shapes as well.
-            self._CompareForward(r2c.astype(np.float32), rank, fft_length,
-                                 use_placeholder=True)
-            self._CompareBackward(c2r.astype(np.complex64), rank, fft_length,
-                                  use_placeholder=True)
+  def testFftLength(self):
+    if test.is_gpu_available(cuda_only=True):
```
Comment (rryan) [inline tensorflow/python/kernel_tests/fft_ops_test.py]: Everything looks fine in fft_ops.cc and fft_ops_test.py except this line which was removed in:
https://github.com/tensorflow/tensorflow/commit/51ec66e4110dbbd37495303bc195b99a5d262e7b

Comment (martinwicke) [inline tensorflow/python/kernel_tests/fft_ops_test.py]: You are saying this line should be removed?

Comment (rryan) [inline tensorflow/python/kernel_tests/fft_ops_test.py]: Yep, this test (and all others in this file) should pass on both GPU and CPU kernels.

**Pull request merged**